### ISSUE TITLE
graphql: `null` sometimes returned for status, gasUsed

### ISSUE THREAD

**Issue by imnotanoob** [ethereum/go-ethereum#26559]
Hello,

I have not noticed this on testnet or prod yet, but there seems to be a schema difference between dev and other environments.
For example with this graphQL Query 

```
{"query":"{block(number: 151) { number hash transactions {from{address} to{address} hash index gas gasUsed effectiveGasPrice logs { topics index data account {address}} status createdContract {address}}}}"}

```

We receive this as the response:

```
  "data" : {
    "block" : {
      "number" : 151,
      "hash" : "0xdda745d33d7db0c209f5992f0d7647437425674a97bb49f8faf418d0ddd97f08",
      "transactions" : [ {
        "from" : {
          "address" : "0x679e7cb3466710f73734641dd3bf90669dcd3e99"
        },
        "to" : null,
        "hash" : "0x482f6fa5400f18cb8be2ced414dd1a2c5701dcd2a4e9e72a80bb7cd737385e56",
        "index" : 0,
        "gas" : "0x989680",
        "gasUsed" : 287497,
        "effectiveGasPrice" : "0x1a13b8600",
        "logs" : [ ],
        "status" : null,
        "createdContract" : {
          "address" : "0xe7073d6fc1ed152738b0442a6b800bf2184fe889"
        }
      } ]
    }
  }
}
```

The "status" shows up as null.

**Comment by rjl493456442** [comment]
What's your command to launch Geth?

I try to launch the dev mode geth `geth --dev --graphql  --http console` and status can be resolved correctly.

**Comment by holiman** [comment]
> What's your command to launch Geth?

And what is your geth version?

**Comment by imnotanoob** [comment]
`1.10.24`.

This doesn't happen often. Extremely randomly. For example if you try 100 times it happens once.
Same for `gasUsed`.

**Comment by fjl** [comment]
Pretty sure it's related with https://github.com/ethereum/go-ethereum/issues/25868

**Comment by s1na** [comment]
This is caused by a data race happening here:

https://github.com/ethereum/go-ethereum/blob/4c23fe97c5472f2c0a6c89fcf4e459316110fb0c/graphql/graphql.go#L633

In which multiple transactions race for setting the receipts of a block.

**Comment by s1na** [comment]
Should be fixed in #26965.
